### ISSUE TITLE
Pandas 1.0 nullable data type produces ArrowTypeError

### ISSUE THREAD

**Issue by ged-steponavicius** [JDASoftwareGroup/kartothek#263]
### Problem description

pyarrow pa.Schema.from_pandas does not work with new pandas 1.0 nullable integer type which breaks store_dataframes_as_dataset

### Example code 
```
import pandas as pd
from functools import partial
from tempfile import TemporaryDirectory
from storefact import get_store_from_url
from kartothek.io.eager import store_dataframes_as_dataset,update_dataset_from_dataframes

dataset_dir = TemporaryDirectory()

store_factory = partial(get_store_from_url, f"hfs://{dataset_dir.name}")
store = get_store_from_url( f"hfs://{dataset_dir.name}")

df = pd.DataFrame([{'int_col':1},{'int_col':2}])
df['int_col'] = df['int_col'].astype(pd.Int64Dtype())

dm = store_dataframes_as_dataset(
    store_factory,
    "sample_ds", 
    [df]
    )

```

error: ArrowTypeError: Did not pass numpy.dtype object

schema = pa.Schema.from_pandas(df) - does not work
schema = pa.Table.from_pandas(df).schema - works fine

pyarrow 0.16
kartothek 3.8.1

**Comment by lr4d** [comment]
Thanks for the report @ged-steponavicius . I see you also opened this issue in the Arrow JIRA tracker, I'll just link that here for posterity https://issues.apache.org/jira/browse/ARROW-8223

**Comment by xhochy** [comment]
This is a duplicate of https://github.com/JDASoftwareGroup/kartothek/issues/234 and will be fixed in the soon-to-be-released `pyarrow=0.17.0` and there is already a failing unit test in https://github.com/JDASoftwareGroup/kartothek/blob/5cba248cbfe9f7375ea3656850bbc98c7fb1746c/kartothek/io/testing/read.py#L733-L762 (it doesn't fail anymore with the Arrow nightlies).

**Comment by lr4d** [comment]
Thanks for the input @xhochy , I'll close this for now
